tests/integration/test_advanced_indexing.py

```diff
@@ -81,6 +95,95 @@ def test_setitem_scalar_1d(arr, idx, val):
     assert np.array_equal(arr, [-1])
 
 
+@pytest.mark.parametrize("idx", IDXS_EMPTY_1D)  # idx = []
+@pytest.mark.parametrize("arr", ARRS_EMPTY_1D)  # arr = [42], [5], []
+def test_getitem_empty_1d(arr, idx):
+    assert np.array_equal(arr[idx], [])
+
+
+@pytest.mark.parametrize("idx", IDXS_EMPTY_1D)  # idx = []
+@pytest.mark.parametrize("arr", ARRS_EMPTY_1D)  # arr = []
+@pytest.mark.parametrize("val", VALS_EMPTY_1D)  # val = []
+def test_setitem_empty_1d(arr, idx, val):
+    arr[idx] = val
+    assert np.array_equal(arr[idx], [])
+
+
+def mk_deferred_array(lib, shape):
+    if np.prod(shape) != 0:
+        return lib.ones(shape)
+    # for shape (2,0,3,4): good_shape = (2,1,3,4)
+    good_shape = tuple(max(1, dim) for dim in shape)
+    # for shape (2,0,3,4): key = [:,[False],:,:]
+    key = tuple([False] if dim == 0 else slice(None) for dim in shape)
+    return lib.ones(good_shape)[key]
+
+
+def gen_args():
+    result = []
+    for arr_ndim in range(1, LEGATE_MAX_DIM + 1):
+        for idx_ndim in range(1, arr_ndim + 1):
+            for zero_dim in range(arr_ndim):
+                result += [[arr_ndim, idx_ndim, zero_dim]]
+    return result
```

Comment:
FYI this should also work:
```suggestion
def gen_args():
    for arr_ndim in range(1, LEGATE_MAX_DIM + 1):
        for idx_ndim in range(1, arr_ndim + 1):
            for zero_dim in range(arr_ndim):
                yield arr_ndim, idx_ndim, zero_dim
```

Reply:
Thank you!